loggable.js

```diff
@@ -0,0 +1,26 @@
+function loggable(arg1, arg2) {
+  if (typeof arg1 === 'string') {
+    arg2.displayName = arg1
+    return loggableFn(arg2)
+  } else if (typeof arg1 === 'function') {
+    return loggableFn(arg1)
+  } else {
+    throw Error('First argument must be a function or string')
+  }
+}
+
+function loggableFn(fn) {
```

Comment:
@pnaza do you think `loggable` is a good name? I could imagine making and independent decorator called `loggable` that just adds logging to macro-style functions (not using pipe at all)

Reply:
I can't really think of a good name. Maybe `displayFunction` `logFunction`

Reply:
I think I meant to put this comment on another line. I'm talking about the exported function `loggable` (not `loggableFn` on this line). I don't know of a good name either.

I was thinking about making another repo where it exported a `loggable` or `addLogging` function that wraps either normal functions or decorates object methods with Cypress Command Log boilerplate. For example:

```ts
// Add all the Cypress.log boilerplate. `makeLoggable` requires and output type of `Cypress.Chainable<any>`
export const getList = makeLoggable('getList', () => {
  return cy.get('.todo-list')
})

export const PageObject = {
  @makeLoggable() // don't have to supply the name, it is inferable
  getList() {
    return cy.get('.todo-list')
  }
}
```

The Cypress log would look like:
```
GETLIST  ()
```

This would be probably in direct competition to `cy.pipe`. Perhaps maybe not for starter-style helpers that are not inside a `.pipe`. That would allow us to have logging without having a `cy.get('body').pipe` to start us off.